ambry-tools/src/main/java/com.github.ambry/tools/perf/rest/NettyPerfClient.java

```diff
@@ -513,7 +532,13 @@ private void reset() {
         request.headers().add(RestUtils.Headers.TARGET_ACCOUNT_NAME, targetAccountName);
         request.headers().add(RestUtils.Headers.TARGET_CONTAINER_NAME, targetContainerName);
       } else {
-        request = new DefaultFullHttpRequest(HttpVersion.HTTP_1_1, HttpMethod.GET, path);
+        if (pathList == null) {
+          request = new DefaultFullHttpRequest(HttpVersion.HTTP_1_1, HttpMethod.GET, path);
+        } else {
+          counter.compareAndSet(pathList.size(), 0);
```

Comment:
This could be a bit flaky, like if the current value is `pathList.size() - 1` and all threads are about to run line 539. If they all execute before the loops get back to line 538, the value of counter could be greater than than `pathList.size()`.

Instead of this you could consider using a modulo on line 540 or follow some of the solutions here: https://stackoverflow.com/questions/7572434/how-to-implement-a-concurrent-circular-ticker-counter-in-java

Reply:
Good point! Thanks!

In our case, I think we can simply use `counter.getAndIncrement() % pathList.size()`. This is not a true circular ticker if `Int.max % pathList.size() != 0`, but this works and we don't require a true  circular ticker.